OtlTaskControl.pas

```diff
@@ -1656,18 +1682,52 @@ destructor TOmniTaskExecutor.Destroy;
   FreeAndNil(oteTerminateHandles);
   FreeAndNil(oteMethodHash);
   FreeAndNil(oteException);
-  DSiCloseHandleAndNull(oteCommRebuildHandles);

```

Comment:
It is still there. It is just hidden by compiler magic. oteCommRebuildHandles is now an interface pointer of type IOmniEvent. When the TOmniTaskExecutor object is destroyed, the interface data members are de-referenced, and when the reference count reaches zero, the event resource is then de-allocated.